### PR TITLE
Add a new country question

### PR DESCRIPTION
**Pull request by inigomarquinez**
## Description

Instead of having to maintain a local updated list of countries, we make use of [this library](https://annexare.github.io/Countries/) to get the countries data (including flags) and [this library](https://github.com/michaelwittig/node-i18n-iso-countries#readme) to do the translations to different locales.

Apart from that, the new country question has some new features (that should be added inside the `config` property):

- `whitelist` to show only some countries
- `blacklist` to remove some countries from the list
- `flag` to show the country flag along with the name
- `search` to be able to search by typing in the input
- `priorityOptions` to show specific countries at the beginning of the list

Just to keep in mind:

- The language to display the countries will be the one used when setting the `language` property in the FormBuilder. In future releases, we'll include a local config option to be able to override it and even show the countries in their native language.
- The countries will be sorted by name in the local language (except those in the priority list)

Other changes:
- update some dependencies
- improve the example app to support multiple forms
- this is the first question that makes use of [`useFormContext`](https://react-hook-form.com/docs/useformcontext) to avoid having to send it from parent to children. To make it work we had to wrap the form under the [`FormProvider`](https://react-hook-form.com/docs/formprovider).

Still pending (for future PR):
- Upgrade rollup to latest version
- Fix vulnerabilities
- Add unit tests for the new question using react testing library

## Motivation and Context

Improve the country question to have an updated list of countries

## How Has This Been Tested?

Adding some examples and running them locally (still pending to add unit tests)

## Screenshots (if appropriate):

https://github.com/onebeyond/react-form-builder/assets/25435858/2058b5dc-e54d-4a29-bff8-0d951fb420d7

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: example/.eslintrc.js [hunk 1]
```diff
@@ -223,6 +223,8 @@ module.exports = exports = {
     'spaced-comment': [WARN, 'always'],
     'import/no-duplicates': WARN,
     'react/jsx-no-bind': WARN,
-    'wrap-regex': WARN
+    'wrap-regex': WARN,
+
```
Comment (NuriaExtremadouro) [inline example/.eslintrc.js:227]: nit: not important, but there's a weird blank line here, I think.
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -172,7 +171,7 @@ Reminder: the 'countryAndRegions' prop that can be sent in the ReactFormBuilder
 ```
 
 ### Custom list country example:
-```yaml
+```json
```
Comment (NuriaExtremadouro) [inline README.md:174]: nit: I think this might be `js` or `jsx`, based on the lack of `""`.

File: src/builder.js [hunk 3]
```diff
@@ -321,7 +336,7 @@ const FormBuilder = ({
   }
 
   return (
-    <>
+    <FormProvider {...useFormObj}>
```
Comment (pabloelisseo) [inline src/builder.js:339]: Why do we need the `FormProvider`? just curious

**Review by pabloelisseo — DISMISSED** [review]
Amazing work here! 🚀

**Review by NuriaExtremadouro — APPROVED** [review]
Left some small comments, but nothing too pressing

File: example/src/error-page.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import { useRouteError } from 'react-router-dom'
+
+export default function ErrorPage() {
+  const error = useRouteError();
+  // console.error(error);
```
[the remark this answers is not in the document]
Comment (inigomarquinez) [reply example/src/error-page.jsx]: Solved [here](https://github.com/onebeyond/react-form-builder/commit/bddcea462988cfa28833df9e2df7f7599be6e8d0)

File: README.md (hunk 2 above)
Comment (inigomarquinez) [reply README.md:175]: Solved [here](https://github.com/onebeyond/react-form-builder/commit/bddcea462988cfa28833df9e2df7f7599be6e8d0)

**Pull request merged**